src/pandas_profiling/model/typeset.py

```diff
@@ -0,0 +1,231 @@
+import imghdr
+import os
+import warnings
+from urllib.parse import urlparse
+
+import pandas as pd
+import visions
+from pandas.api import types as pdt
+from visions import VisionsBaseType, VisionsTypeset
+from visions.relations import IdentityRelation, InferenceRelation
+from visions.utils import nullable_series_contains
+
+from pandas_profiling.config import config
+from pandas_profiling.model.typeset_relations import (
+    category_is_numeric,
+    category_to_numeric,
+    numeric_is_category,
+    series_is_string,
+    string_is_bool,
+    string_to_bool,
+    to_bool,
+    to_category,
+)
+
+
+class PandasProfilingBaseType(VisionsBaseType):

```

Comment:
There's no need to extend the VisionBaseType for PP if it's not leveraging a standardized set of attributes. I would replace `PandasProfilingBaseType` with `VisionsBaseType` and have `Unsupported` inherit directly from `visions.Generic`.